tools/grab_micro_posts.py

```
@@ -108,11 +124,14 @@
     post['original_url'] = post_url
     post['note_id'] = note_id
 
-    with open(file_path, 'w', encoding='utf-8') as f:
-        f.write(frontmatter.dumps(post))
-
-    print(f"Created new post: {file_path} with Note ID: {note_id}")
-    return True
+    try:
+        with open(file_path, 'w', encoding='utf-8') as f:
+            f.write(frontmatter.dumps(post))
+        logging.info(f"Created new post: {file_path} with Note ID: {note_id}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)